components/event-publisher/org.wso2.carbon.event.publisher.core/src/main/java/org/wso2/carbon/event/publisher/core/internal/util/EventPublisherConfigurationBuilder.java

```diff
@@ -204,6 +208,7 @@ public static EventPublisherConfiguration getEventPublisherConfiguration(
             }
         }
 
+        dynamicProperties = resolveSecrets(dynamicProperties);
```

Comment:
Instead of this, can't we resolve the value at the time we read it? for example if the value is encrypted we decrypt it before we add the property to dynamic properties?

Reply:
if so we have to create factory for every time
https://github.com/wso2/carbon-analytics-common/pull/800/files#diff-46d6831a013f38639d810d2744599d04d35e925eaf78ab00631801962ff128b6R294